common/src/Model/Brush.cpp

```diff
@@ -227,6 +227,58 @@ namespace TrenchBroom {
             }
         }
 
+        static const BrushFace* findBestMatchingFace(const BrushFace& face, const std::vector<const BrushFace*>& candidates) {
+            if (candidates.empty()) {
+                return nullptr;
+            }
+
+            // First, look for coplanar candidates
+            std::vector<const BrushFace*> coplanarCandidates;
+            for (auto& candidate : candidates) {
+                if (candidate->onPlane(face.boundary())) {
+                    coplanarCandidates.push_back(candidate);
+                }
+            }
+
+            if (!coplanarCandidates.empty()) {
+                // Return the largest coplanar face
+                return *std::max_element(std::begin(coplanarCandidates), std::end(coplanarCandidates),
+                    [](const BrushFace* lhs, const BrushFace* rhs){
+                        return lhs->area() < rhs->area();
+                    });
+            }
+
+            // No coplanar faces. Return the one with the smallest "face center off reference plane" distance.
+            const auto faceCenterOffPlaneDist = [&](const BrushFace* candidate) -> FloatType {
```

Comment:
Out of interest, why denote the return type of the lambda here?

Reply:
My reasoning is, it's quite difficult to tell what this lambda returns if it's not annotated; it would be whatever `point_distance` returns.

Also, because I assign the lambda to a variable, I think of it like a local function. Functions always get explicit return types so that it's possible to use/reason about them without looking at the implementation. The same applies here.

Reply:
I see! Thanks for explaining.